src/main/org/apache/tools/zip/ExtraFieldUtils.java

```diff
@@ -81,10 +87,10 @@ public static void register(Class<?> c) {
      * @since 1.1
      */
     public static ZipExtraField createExtraField(ZipShort headerId)
-        throws InstantiationException, IllegalAccessException {
+            throws InstantiationException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
```

Comment:
this changes the signature of a public method, I'm afraid we can not do this without breaking backwards compatibility.

Reply:
thanks, missed that, wrapped in InstantiationException.